src/main/src/main/java/org/geoserver/catalog/StoreInfo.java

```diff
@@ -170,4 +171,7 @@
      */
     // <T extends Resource> Iterator<T> getResources(ProgressListener monitor) throws IOException;
 
+    public Date getDateModified();
```

Comment:
Would it not make sense to have this method sin Catalog info? Return NULL by default?

Reply:
> A few general comments:
> 
> 
> I think that hierarchy of objects can be used to reduce the repeated boilerplate code.
> 
> 
> Default methods shoudl be used in interfaces to reduce the need of boilerplate implementations methods and to maintain backwards compatibility.
> 

Understood, will move the methods to CatalogInfo

> 
> Is it real worth the trouble of having the events based approach implementation? Why not updating the timestamps directly in the catalog? It needs of be changed anyway to add the events firing.
> 

Well initially that's how it was implemented for layers but then as things moved forward, it did not seem right when we can listen to catalog events and implement cleanly.

> 
> Why not adding direct attributes for modified timestamp and created timestamp? Using the metadata map for something that's core looks clumsy.
> 

Understood, so all implementations of CatalogInfo will get these fields? That's round about 23 sub-classes 

> 
> I found  behavior that is worth thinking about, if I open a layer (or something else) and don't change anything but click on save the modified timestamp will be updated, this doesn't look right. Update: this looks like something difficult to implement, so probably keep it for a later improvement 😄

understood, a comparison can be made inside the modify event.

> Last comment, since we add the times stamp for everything an is a two or three lines change, would it no make since add it too for workspaces? Ideally it shoudl also listen on workspace setting modifications.

Sure why not :D